sql/parser/eval.go

```diff
@@ -525,45 +634,47 @@ type cmpArgs struct {
 	rightType reflect.Type
 }
 
+var cmpOpResultType = reflect.New(reflect.TypeOf(cmpOps).Elem().Out(0)).Elem().Interface().(Datum)
```

Comment:
All that reflect stuff so you can change the type of the return value for the `cmpOps` functions without adjusting this type. Doesn't seem worth it to me. I challenge anyone who is not well versed in `reflect` to understand this.